File: 409/409.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        int result = 0;
        std::unordered_set<char> single;
        for(char c: s){
            if(single.find(c)!= single.end()){
                single.erase(c);
                result += 2;
            }else{
                single.insert(c);
            }
        }
        if(!single.empty()){
            result++;
        }

        return result;
    }
};

File: 409/409.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function (s) {
  let single = [];
  let result = 0;
  for (let char of s) {
    const index = single.indexOf(char);
    if (index !== -1) {
      single.splice(index, 1);
      result += 2;
    } else {
      single.push(char);
    }
  }
  return single.length >= 1 ? result + 1 : result;
};
